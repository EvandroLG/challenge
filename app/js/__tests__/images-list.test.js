import apiMock from '../__mocks__/api';
import ImagesList from '../images-list';
import http from '../http';

describe('images-list', () => {
  beforeEach(() => {
    http.get = jest.fn(() => {
      return new Promise(resolve => resolve(apiMock));
    });

    document.body.innerHTML = `
      <ul id="js-list">
      </ul>
    `;
  });

  it('should update list html with the datas provided by the api', async () => {
    const imagesList = new ImagesList();
    await imagesList.update('cat');

    expect(http.get).toHaveBeenCalled();
    expect(
      document.getElementById('js-list').innerHTML
    ).toMatchSnapshot();
  });

  it('should not update list html when api returns empty data', async () => {
    http.get = jest.fn(() => {
      return new Promise(resolve => resolve({
        data: [],
      }));
    });

    const imagesList = new ImagesList();
    await imagesList.update('cat');

    expect(
      document.getElementById('js-list').querySelectorAll('li').length
    ).toEqual(0);
  });

  it('should request and display more images', async () => {
    const imagesList = new ImagesList();
    await imagesList._updateNextPage();
    await imagesList._updateNextPage();

    expect(imagesList.page).toEqual(2);
    expect(http.get).toHaveBeenCalled();
    expect(
      document.getElementById('js-list').innerHTML
    ).toMatchSnapshot();
  });

  it('should not display more images when request returns empty data', async () => {
    http.get = jest.fn(() => {
      return new Promise(resolve => resolve({
        data: [],
      }));
    });

    const imagesList = new ImagesList();
    await imagesList._updateNextPage();

    expect(imagesList.page).toEqual(0);
    expect(
      document.getElementById('js-list').querySelectorAll('li').length
    ).toEqual(0);
  });
});
