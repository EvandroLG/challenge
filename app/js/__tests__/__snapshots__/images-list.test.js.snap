// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`images-list should request and display more images 1`] = `
"
            
                <li>
                    <img src=\\"https://media2.giphy.com/media/3o7aTIrTw1ckKbqra8/giphy.gif\\" width=\\"360\\" height=\\"270\\">
                </li>
            
                <li>
                    <img src=\\"https://media0.giphy.com/media/3o7aTsNEoU0k75RWKI/giphy.gif\\" width=\\"480\\" height=\\"270\\">
                </li>
            
                <li>
                    <img src=\\"https://media2.giphy.com/media/3o7aTIrTw1ckKbqra8/giphy.gif\\" width=\\"360\\" height=\\"270\\">
                </li>
            
                <li>
                    <img src=\\"https://media0.giphy.com/media/3o7aTsNEoU0k75RWKI/giphy.gif\\" width=\\"480\\" height=\\"270\\">
                </li>
            
                <li>
                    <img src=\\"https://media2.giphy.com/media/3o7aTIrTw1ckKbqra8/giphy.gif\\" width=\\"360\\" height=\\"270\\">
                </li>
            
                <li>
                    <img src=\\"https://media2.giphy.com/media/3o7aTIrTw1ckKbqra8/giphy.gif\\" width=\\"360\\" height=\\"270\\">
                </li>
            
                <li>
                    <img src=\\"https://media0.giphy.com/media/3o7aTsNEoU0k75RWKI/giphy.gif\\" width=\\"480\\" height=\\"270\\">
                </li>
            
                <li>
                    <img src=\\"https://media2.giphy.com/media/3o7aTIrTw1ckKbqra8/giphy.gif\\" width=\\"360\\" height=\\"270\\">
                </li>
            
                <li>
                    <img src=\\"https://media0.giphy.com/media/3o7aTsNEoU0k75RWKI/giphy.gif\\" width=\\"480\\" height=\\"270\\">
                </li>
            
                <li>
                    <img src=\\"https://media2.giphy.com/media/3o7aTIrTw1ckKbqra8/giphy.gif\\" width=\\"360\\" height=\\"270\\">
                </li>
            "
`;

exports[`images-list should update list html with the datas provided by the api 1`] = `
"
                <li>
                    <img src=\\"https://media2.giphy.com/media/3o7aTIrTw1ckKbqra8/giphy.gif\\" width=\\"360\\" height=\\"270\\">
                </li>
            
                <li>
                    <img src=\\"https://media0.giphy.com/media/3o7aTsNEoU0k75RWKI/giphy.gif\\" width=\\"480\\" height=\\"270\\">
                </li>
            
                <li>
                    <img src=\\"https://media2.giphy.com/media/3o7aTIrTw1ckKbqra8/giphy.gif\\" width=\\"360\\" height=\\"270\\">
                </li>
            
                <li>
                    <img src=\\"https://media0.giphy.com/media/3o7aTsNEoU0k75RWKI/giphy.gif\\" width=\\"480\\" height=\\"270\\">
                </li>
            
                <li>
                    <img src=\\"https://media2.giphy.com/media/3o7aTIrTw1ckKbqra8/giphy.gif\\" width=\\"360\\" height=\\"270\\">
                </li>
            "
`;
